engine/class_modules/sc_death_knight.cpp

```diff
@@ -3149,6 +3273,11 @@ struct melee_t : public death_knight_melee_attack_t
       {
         trigger_bloodworm();
       }
+
+      if ( p() -> legendary.reanimated_shambler -> ok() )
```

Comment:
Does it only trigger on melee auto attacks?

Reply:
Oh duh I need to move it up to death_knight_melee_attack_t.  Forgot bloodworms was autos only and thought melee_t was a part of every ability that involve a melee hit.  Abilities can proc it too, based on the spelldata and what I'm seeing in game.  

`Proc Flags       : ..x.x... ........ ........ ........ 
                 : White Melee, Yellow Melee`

Reply:
Proc numbers moved a bit after the change

Before
`Reanimated Shambler | 8.2 | 3.0 | 20.0 | 32.9s | 3.4s | 133.7s`

After
`Reanimated Shambler | 8.2 | 2.0 | 22.0 | 32.5s | 0.1s | 130.0s`

Reply:
Proc numbers moved a bit after the change

Before
Reanimated Shambler | 8.2 | 3.0 | 20.0 | 32.9s | 3.4s | 133.7s

After
Reanimated Shambler | 8.2 | 2.0 | 17.0 | 33.0s | 0.1s | 135.6s

I'm not 100% on the accuracy of putting it in execute as it would probably be more technically correct to make sure the attack isn't parried.  But impact gets run multiple times for AoE so shouldn't have it there either.  Unless each hit of SS cleave has a chance to proc it...

Reply:
It should be implemented like a regular proc : simc handles this by registering special effects and callbacks with specific proc flags and actions/buffs to execute.